packages/editor/src/browser/editor-manager.ts

```diff
@@ -105,6 +112,28 @@ export class EditorManagerImpl implements EditorManager, WidgetFactory {
         });
     }
 
+    /*
+     *  check if a widget exists already and rename it
+     */
+    renameWidget(factoryId: string, options: any, newOptions: any, newLabel: any): void {
+        let key = this.widgetManager.toKey({ factoryId, options });
+        let existingWidget = this.widgetManager.getWidgetsMap().get(key);
+        if (existingWidget) {
+            existingWidget.title.label = newLabel;
+            this.updateWidgetKey(key, existingWidget, factoryId, options, newOptions);
+        }
```

Comment:
This is just externalizing your original implementation into the widget manager.
You should use EditorManager methosd to get to your widget.